### PR TITLE
CRM-21364 Non controversial fixes for ONLY_FULL_GROUP_BY and NO_ZERO_…

### PR DESCRIPTION
**Pull request by seamuslee001**
…DATE sqlModes

Overview
----------------------------------------
This pulls out the non controversial fixes from https://github.com/civicrm/civicrm-core/pull/11053/files and https://github.com/civicrm/civicrm-core/pull/11207. ping @monishdeb @eileenmcnaughton @colemanw 

This aims to fix some test failures on the unbuntu1604 matrix

### CODE REVIEW

**Review by monishdeb — APPROVED** [review]
Reviewed and tested the patch, works fine for me.  @eileenmcnaughton can you review and merge the code, esp. the changes made in  CRM/Activity/BAO/Activity.php that I suggested earlier.

**Comment by seamuslee001** [comment]
@eileen are you ok with this?

**Comment by seamuslee001** [comment]
@monishdeb tests have passed

**Comment by monishdeb** [comment]
Happy with the final patch.. merging now

File: CRM/Activity/BAO/Activity.php [hunk 1]
```diff
@@ -909,14 +909,20 @@ public static function deprecatedGetActivities($input) {
...
+    $insertValueSQL = $selectColumns = array();
     // The activityTempTable contains the sorted rows
     // so in order to maintain the sort order as-is we add an auto_increment
     // field; we can sort by this later to ensure the sort order stays correct.
     $sql .= " fixed_sort_order INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,";
     foreach ($tableFields as $name => $desc) {
       $sql .= "$name $desc,\n";
       $insertValueSQL[] = $name;
+      if ($name == 'source_contact_name' && CRM_Utils_SQL::supportsFullGroupBy()) {
+        $selectColumns[] = "ANY_VALUE(tbl.$name)";
```
Comment (mattwire) [inline CRM/Activity/BAO/Activity.php]: @seamuslee001 ANY_VALUE does not exist on mariadb (https://jira.mariadb.org/browse/MDEV-10426) so including it here prevents us using mariadb.  Just hit this while saving a new membership

**Pull request merged**